FHEM/00_SIGNALduino.pm

```diff
@@ -884,8 +886,8 @@ sub SIGNALduino_Set_LaCrossePairForSec {
 	my ($hash, @a) = @_;
 
 	#              set NAME                a[0]              a[1]             a[2]
-	return "Usage: set $hash->{NAME} $a[0] <seconds_active> [ignore_battery " if(!$a[0] || $a[1] !~ m/^\d+$/ || ($a[2] && $a[2] ne "ignore_battery") );
-	$hash->{LaCrossePair} = 1;  # LaCrosse autoCreateState: 0 = autoreate not defined | 1 = autocreate defined | 2 = autocreate active
+	return "Usage: set $hash->{NAME} $a[0] <seconds_active> [ignore_battery]" if(!$a[0] || $a[1] !~ m/^\d+$/ || ($a[2] && $a[2] ne "ignore_battery") );
```

Comment:
Der Befehl stand vorher schon da, weiss ich, jetzt muss ich mir das aber mal ansehen:

`f(!$a[0] || $a[1] !~ m/^\d+$/ || ($a[2] && $a[2] ne "ignore_battery") )`

Welcher Wert soll mit !$a[0] denn ausgeschlossen werden? Vermutlich der Wert `0` oder auch "" Was hältst Du davon, wenn wir die Variablen gleich mit einer erklärenden Bezeichnung abrufen, also in der Form:
```
my $comand = shift;
my $seconds = shift // 2; # Vielleicht macht ein default sinn, wenn kein Wert übergeben wird
my $ign_battery = shift;
```
Erfassen.
Dann zu ` $a[1] !~ m/^\d+$/ `  wäre besser ` $a[1] !~ m/^[0-9]+$/ ` Wobei das quasi unendlich lange zulässt. 

Der letzte Abschnitt:
`($a[2] && $a[2] ne "ignore_battery")` 

Könnte zu
`( defined $a[2] && $a[2] ne "ignore_battery")` 
werden.

Reply:
1)
> Welcher Wert soll mit !$a[0] denn ausgeschlossen werden? Vermutlich der Wert 0 oder auch "" Was hältst Du davon, wenn wir die Variablen gleich mit einer erklärenden Bezeichnung abrufen, also in der Form:

>my $comand = shift;
my $seconds = shift // 2; # Vielleicht macht ein default sinn, wenn kein Wert übergeben wird
my $ign_battery = shift;

Ich finde es ein wenig unsinnig, wenn ich in der kleinen Sub 
https://github.com/HomeAutoUser/RFFHEM/blob/dev-r35_xFSK_TEST/FHEM/00_SIGNALduino.pm#L884-L895
nur für einmal die Variablen zu definieren. Ja, mit diesem vorgehen erhält man einen besseren Einblick aber in dem Kommentar oben drüber ist dies auch ersichtlich.
https://github.com/HomeAutoUser/RFFHEM/blob/dev-r35_xFSK_TEST/FHEM/00_SIGNALduino.pm#L888-L889
OK, etwas verschoben in der Einrückung aber es ist zuordenbar.

2)
>Dann zu $a[1] !~ m/^\d+$/ wäre besser $a[1] !~ m/^[0-9]+$/ Wobei das quasi unendlich lange zulässt.

Zu deinem Vorschlag, das dies besser wäre ;-)
Du möchtest gern `perlcritic Konform` arbeiten bzw. bemängelst Dinge welche wir diesbezüglich ändern sollten.
Da wiedersprichst du dir in der Zeile selbst. Mache den Test mit `perlcritic` Level (1 - brutal) und du wirst sehen, das deine Schreibweise nicht konform ist. *smile* Eine konforme Schreibweise an der Stelle wäre
`$a[1] !~ m/^\d+$/xms`

3)
>Der letzte Abschnitt:
($a[2] && $a[2] ne "ignore_battery")

>Könnte zu
( defined $a[2] && $a[2] ne "ignore_battery")
werden.

Ja, gern können wir es zu
`( defined $a[2] && $a[2] ne "ignore_battery")`
ändern.